src/config/session.c

```diff
@@ -99,6 +134,9 @@ session_environment_init(const char *dir)
 void
 session_autostart_init(const char *dir)
 {
+	/* Update dbus and systemd user environment, each may fail gracefully */
+	update_activation_env("DISPLAY WAYLAND_DISPLAY XDG_CURRENT_DESKTOP");
```

Comment:
If you'll move the environment variables inside the function, you can do it with no allocations at all.

Reply:
That is what I had at the beginning (2 straight copies of the env vars) but for maintainability reasons I think this is actually cleaner.

If we decide in the future to add more variables we just have to modify this single argument and everything else automatically works.